kernels/volk/volk_64f_x2_dot_prod_64f.h

```diff
@@ -0,0 +1,842 @@
+                                                       const double* taps,
+                                                       unsigned int num_points)
+{
+    unsigned int number;
+    const unsigned int fourthPoints = num_points / 4;
+
+    const double* aPtr = input;
+    const double* bPtr = taps;
+
+    __m256d dotProdVal = _mm256_setzero_pd();
+    __m256d aVal1, bVal1;
+
+    for (number = 0; number < fourthPoints; number++) {
+
+        aVal1 = _mm256_loadu_pd(aPtr);
+        bVal1 = _mm256_loadu_pd(bPtr);
+        aPtr += 4;
+        bPtr += 4;
+
+        dotProdVal = _mm256_fmadd_pd(aVal1, bVal1, dotProdVal);
+    }
+
+    __VOLK_ATTR_ALIGNED(32) double dotProductVector[4];
+    _mm256_storeu_pd(dotProductVector,
+                     dotProdVal); // Store the results back into the dot product vector
+
+    double dotProduct = dotProductVector[0] + dotProductVector[1] + dotProductVector[2] +
+                       dotProductVector[3];
+
+    for (number = fourthPoints * 4; number < num_points; number++) {
+        dotProduct += ((*aPtr++) * (*bPtr++));
+    }
+
+    *result = dotProduct;
+}
+#endif /* LV_HAVE_AVX2 && LV_HAVE_FMA */
+
+#if LV_HAVE_AVX512F
+#include <immintrin.h>
+static inline void volk_64f_x2_dot_prod_64f_u_avx512f(double* result,
+                                                      const double* input,
+                                                      const double* taps,
+                                                      unsigned int num_points)
+{
+    unsigned int number;
+    const unsigned int eighthPoints = num_points / 8;
+
+    const double* aPtr = input;
+    const double* bPtr = taps;
+
+    __m512d dotProdVal = _mm512_setzero_pd();
+    __m512d aVal1, bVal1;
+
+    for (number = 0; number < eighthPoints; number++) {
+
+        aVal1 = _mm512_loadu_pd(aPtr);
+        bVal1 = _mm512_loadu_pd(bPtr);
+        aPtr += 8;
+        bPtr += 8;
+
+        dotProdVal = _mm512_fmadd_pd(aVal1, bVal1, dotProdVal);
+    }
+
+    __VOLK_ATTR_ALIGNED(64) double dotProductVector[8];
+    _mm512_storeu_pd(dotProductVector,
+                     dotProdVal); // Store the results back into the dot product vector
+
+    double dotProduct = dotProductVector[0] + dotProductVector[1] + dotProductVector[2] +
+                       dotProductVector[3] + dotProductVector[4] + dotProductVector[5] +
+                       dotProductVector[6] + dotProductVector[7];
+
+    for (number = eighthPoints * 8; number < num_points; number++) {
+        dotProduct += ((*aPtr++) * (*bPtr++));
+    }
+
+    *result = dotProduct;
+}
+#endif /* LV_HAVE_AVX512F */
+
+#endif /*INCLUDED_volk_64f_x2_dot_prod_64f_u_H*/
+
+#ifndef INCLUDED_volk_64f_x2_dot_prod_64f_a_H
+#define INCLUDED_volk_64f_x2_dot_prod_64f_a_H
+
+#include <stdio.h>
+#include <volk/volk_common.h>
+
+
+#ifdef LV_HAVE_GENERIC
+
+
+static inline void volk_64f_x2_dot_prod_64f_a_generic(double* result,
```

Comment:
I suggest to remove this function. Old kernels have the aligned generic version sometimes. But the generic kernel should not rely on any alignment. Also, this kernel yields wildly differing results compared to the "unaligned".